src/probabilities_estimators/permutation_ordinal/spatial_permutation.jl

```diff
@@ -77,6 +100,43 @@ function SpatialSymbolicPermutation(
     SpatialSymbolicPermutation{D, p, typeof(valid)}(stencil, copy(stencil), arraysize, valid)
 end
 
+function SpatialSymbolicPermutation(
```

Comment:
If the `extent` and `lag` approach is only implemented for 2D and 3D, this must be mentioned in the docstring. 

Even better: we should perhaps consider just use some generated code here to generalize to higher dimensions without code duplication.

Reply:
> Even better: we should perhaps consider just use some generated code here to generalize to higher dimensions without code duplication.

Good point, I had been thinking about that too. I implemented the generalized way now

Reply:
> Good point, I had been thinking about that too. I implemented the generalized way now

Excellent.